File: 240123/1이 3개 이상 있는 위치/place-more-than-3-ones.py
n = int(input())

grid = [
    list(map(int, input().split()))
    for _ in range(n)
]

drs = [0, 1, 0, -1]
dcs = [1, 0, -1, 0]

answer = 0


def in_range(r, c):
    return r >= 0 and r < n and c >= 0 and c < n


def search(r, c):
    count = 0

    for dr, dc in zip(drs, dcs):
        nr, nc = r + dr, c + dc

        if in_range(nr, nc):
            count += grid[nr][nc]

    if count >= 3:
        return 1
    return 0    


for r in range(n):
    for c in range(n):
        answer += search(r, c)

print(answer)

File: 250709/1이 3개 이상 있는 위치/place-more-than-3-ones.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const n = Number(input[0]);
const grid = input.slice(1).map(line => line.split(' ').map(Number));

let answer = 0;

// Please Write your code here.
function countOneBlock(r, c) {
    const [dr, dc] = [[1, -1, 0, 0], [0, 0, 1, -1]];
    let count = 0;

    for (let i = 0; i < 4; i++) {
        const [nr, nc] = [r + dr[i], c + dc[i]];

        if (inRange(nr, nc) && grid[nr][nc] === 1) {
            count++;
        }
    }

    return count;
}

function inRange(r, c) {
    return r >= 0 && r < n && c >= 0 && c < n;
}

function solution() {
    for (let i = 0; i < n; i++) {
        for (let j = 0; j < n; j++) {
            if (countOneBlock(i, j) >= 3) {
                answer++;
            }
        }
    }

    console.log(answer);
}

solution();